src/libraries/Microsoft.Extensions.Caching.Memory/src/CacheEntry.cs

```diff
@@ -11,28 +11,26 @@
 
 namespace Microsoft.Extensions.Caching.Memory
 {
-    internal class CacheEntry : ICacheEntry
+    internal partial class CacheEntry : ICacheEntry
     {
         private static readonly Action<object> ExpirationCallback = ExpirationTokensExpired;
 
-        private readonly object _lock = new object();
         private readonly MemoryCache _cache;
 
-        private IList<IDisposable> _expirationTokenRegistrations;
-        private IList<PostEvictionCallbackRegistration> _postEvictionCallbacks;
-        private IList<IChangeToken> _expirationTokens;
+        private CacheEntryTokens _tokens; // might be null if user is not using the tokens or callbacks
```

Comment:
```suggestion
        private CacheEntryTokens? _tokens; // might be null if user is not using the tokens or callbacks
```
Is nullability enabled in this project?

Reply:
it is not